gateway-messages/src/mgs_to_sp.rs

```diff
@@ -292,6 +293,45 @@ pub enum LedComponentAction {
     Blink,
 }
 
+/// Actions for the Monorail switch
+#[derive(
+    Copy, Clone, Serialize, SerializedSize, Deserialize, PartialEq, Eq, Debug,
+)]
+pub enum MonorailComponentAction {
+    /// Request an `UnlockChallenge`
+    ///
+    /// The SP will reply with the weakest challenge that it will accept, and
+    /// will store that challenge locally (to prevent replay attacks)
+    RequestChallenge,
+
+    /// Unlock the management network, allowing access from the tech port
+    Unlock {
+        challenge: UnlockChallenge,
+        response: UnlockResponse,
+        time_sec: u32,
+    },
+
+    /// Relock the management network
+    Lock,
+}
+
+/// Actions for the Monorail switch
+#[derive(
+    Copy, Clone, Serialize, SerializedSize, Deserialize, PartialEq, Eq, Debug,
+)]
+pub enum UnlockChallenge {
+    /// Unlock given an [UnlockResponse::Trivial]
+    Trivial,
```

Comment:
It would be good to see a nonce here. I know it doesn't affect security, but it does detect challenges that are out of sync with responses, a class of bugs that we could iron out now.

Reply:
Good idea, added a timestamp in 56e4bccdc158 (deliberately calling it `timestamp` instead of `nonce` so no one starts thinking it's cryptographic)